data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/ResponseHandler.java

```diff
@@ -5,20 +5,21 @@
 
 package org.opensearch.dataprepper.peerforwarder.server;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.MediaType;
 
 import java.util.Objects;
 
-public class RequestExceptionHandler {
+/**
+ * Class to handle exceptions while processing HTTP POST requests by {@link PeerForwarderHttpService}
+ * and sends back the HTTP response based on exception
+ * @since 2.0
+ */
+public class ResponseHandler {
 
     public HttpResponse handleException(final Exception e, final String message) {
         Objects.requireNonNull(message);
-        if (e instanceof JsonProcessingException) {
```

Comment:
Right now the only exception is `JsonProcessingException` and I think with current code all `JsonProcessingException` will be Internal Server Errors since client throws an exception if there's any issue with serializing events. I'm not sure how we can differentiate the exception here.

Reply:
When parsing JSON from the client, the server should respond with Bad Request if the server cannot parse that JSON. I'm fine with keeping what you previously had. I'm just not sure it is the ideal class design.